docs/contracts/overview/Token/customise-transfer-behaviour.md

```diff
@@ -0,0 +1,103 @@
+
+```solidity
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+// tokens
+import { _LSP4_TOKEN_TYPE_TOKEN } from "@lukso/lsp4-contracts/contracts/LSP4Constants.sol";
+import { LSP7DigitalAsset } from "@lukso/lsp7-contracts/contracts/LSP7DigitalAsset.sol";
+
+// libraries
+import { LSP2Utils } from "@lukso/lsp2-contracts/contracts/LSP2Utils.sol";
+
+contract MyToken is LSP7DigitalAsset {
+    /// @dev Available options for 4th parameter token type (uint256)
+    /// - 0 for Token
+    /// - 1 for NFT
+    /// - 2 for Collection
+    constructor(
+        string memory name_,
+        string memory symbol_,
+        address contractOwner_,
+        bool isNonDivisible_
+    )
+        LSP7DigitalAsset(name_, symbol_, contractOwner_, _LSP4_TOKEN_TYPE_TOKEN, isNonDivisible_)
+    {
+        // constructor logic ...
+    }
+
+    function _afterTokenTransfer(
+        address from,
+        address to,
+        uint256, /* amount */
+        bytes memory /* data */
+    )
+        internal
+        virtual
+        override
+    {
+        // TokensSentTx:<address>
+        bytes32 tokensSentTxDataKey =
+            LSP2Utils.generateMappingKey({ keyPrefix: bytes10(keccak256("TokensSentTx")), bytes20Value: bytes20(from) });
+
+        // TokensReceivedTx:<address>
+        bytes32 tokensReceivedTxDataKey = LSP2Utils.generateMappingKey({
+            keyPrefix: bytes10(keccak256("TokensReceivedTx")),
+            bytes20Value: bytes20(to)
+        });
+
+        bytes memory tokensSentTxValue = _getData(tokensSentTxDataKey);
+        bytes memory tokensReceivedTxValue = _getData(tokensReceivedTxDataKey);
+
+        // sanity check to ensure we can abi-decode correctly
+        require(tokensSentTxValue.length == 32, "Invalid uint256 encoded value under `TokensSentTx:<address> data key");
+        require(
+            tokensReceivedTxValue.length == 32,
+            "Invalid uint256 encoded value under `TokensReceivedTx:<address> data key"
+        );
+
+        uint256 tokensSentTxCount = abi.decode(tokensSentTxValue, (uint256));
+        uint256 tokensReceivedTxCount = abi.decode(tokensReceivedTxValue, (uint256));
+
+        // increment the counter + set data in the storage
+        tokensSentTxCount++;
+        tokensReceivedTxCount++;
+
+        _setData(tokensSentTxDataKey, abi.encode(tokensSentTxCount));
+        _setData(tokensReceivedTxDataKey, abi.encode(tokensReceivedTxCount));
+    }
+}
+
+```
+
+This example is minimalist and only stores a counter as a number. Still, any info related to the token transfer could be stored during the transfer (_e.g: the amount, the data passed, the gas price, the balance before and after, etc..._). This way, the storage of the token contract can act, for instance, as: 💡
```

Comment:
Not sure if it was added by mistake or not but what's the lightbulb doing there? 😄 
```suggestion
This example is minimalist and only stores a counter as a number. Still, any info related to the token transfer could be stored during the transfer (_e.g: the amount, the data passed, the gas price, the balance before and after, etc..._). This way, the storage of the token contract can act, for instance, as:
```

Reply:
eagle eyes again! haha